### PR TITLE
Refactor rest.FinishRequest function 

### PR DESCRIPTION
**Pull request by tkashem**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Refactor `rest.FinishRequest` function:
- move the function to its own package as is
- use a single channel to propagate the result from the child goroutine to the parent.

This will pave the way for https://github.com/kubernetes/kubernetes/pull/97428

#### Special notes for your reviewer:
The first commit (move FinishRequest to its own package ) is a move as is without any change -  https://github.com/kubernetes/kubernetes/pull/100523/commits/393a1f73fbaa1bf1facb8882eaf4fead16b94f58 

So if you look at the second commit, you will see the changes introduced by the refactor.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sttts** [comment]
/approve
/lgtm
/hold

@p0lyn0mial remove the hold when are happy with the PR

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/finisher/finisher.go [hunk 1]
```diff
@@ -30,52 +30,79 @@ import (
...
+	// object stores the response returned by the ResultFunc function
+	object runtime.Object
+	// err stores the error returned by the ResultFunc function
+	err error
+	// reason stores the reason from a panic thrown by the ResultFunc function
+	reason interface{}
+}
+
+// Return processes the result returned by a ResultFunc function
+func (r *result) Return() (runtime.Object, error) {
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/finisher/finisher.go:44]: this could be a private method

**Pull request merged**